### DISCUSSION TITLE
How far has USDZ compression come?

### DISCUSSION THREAD

**Discussion by phreak** [google/model-viewer#4318] [General]
Hi all,

I am working with GLBs and USDZs, and while DRACO compression is an excellent choice to compress GLBs, I have little experience with USDZ compression. USDZs are created separately, as I can not rely on model quality conversion through the model viewer (USDZ handles shaders and textures a little differently sometimes).

For now, my workflow is that I use Apple's Reality Composer to compress USDZ's. But that only work on single file export and not batches of models.

What's your experience on this?
How do you tackle the issue?

**Comment by elalish** [comment]
My experience working with USDZ is that's it's not much fun. Our auto-converter is the solution we have; if you find better ones, please share!

**Comment by phreak** [comment]
@elalish: Thank you for your return on my question. Can you tell me what happens in regards of the auto-converter when the GLB comes with WebP textures? Are they converted?

I ask because USDZ is very difficult to bring down in file size, in opposition to GLB. With many texture and AO, USDZ can be easily 5-8x the file size of a GLB file. Therefore I ask myself what is easier to work with, optimize and deliver a separate USDZ or "trust" in the auto-converter.

**Reply by elalish** [reply]
You should try it! It depends very much on how you author your models, but for the common workflows, it works well. Just load your GLB in our [editor](https://modelviewer.dev/editor/) and click "Deploy Mobile", scan the QR code with your iPhone and enter AR - you'll see the result of USDZ auto-conversion and you can verify the quality for yourself. No animation support yet, sadly!

**Reply by Muhammad-Awais-Awan** [reply]
isn't this editor is default google editor from model viewer library?
also if its your own so how you are converting this? by blender cli or someother means?
we also built our own converter but there are some issues in some scenarios in it
